src/lib/seam/connect/seam-http-request.ts

```diff
@@ -0,0 +1,127 @@
+import { serializeUrlSearchParams } from '@seamapi/url-search-params-serializer'
+import type { Method } from 'axios'
+
+import type { Client } from './client.js'
+import type { SeamHttpRequestOptions } from './options.js'
+import { resolveActionAttempt } from './resolve-action-attempt.js'
+import { SeamHttpActionAttempts } from './routes/index.js'
+
+interface SeamHttpRequestParent {
+  readonly client: Client
+  readonly defaults: Required<SeamHttpRequestOptions>
+}
+
+export interface SeamHttpRequestConfig<TBody, TResponseKey> {
+  readonly path: string
+  readonly method: Method
+  readonly body?: TBody
+  readonly query?: undefined | Record<string, unknown>
+  readonly responseKey: TResponseKey
+  readonly options?: Pick<SeamHttpRequestOptions, 'waitForActionAttempt'>
+}
+
+export class SeamHttpRequest<
+  const TBody,
+  const TResponse,
+  const TResponseKey extends keyof TResponse | undefined,
+> implements
+    PromiseLike<
+      TResponseKey extends keyof TResponse ? TResponse[TResponseKey] : undefined
+    >
+{
+  readonly #parent: SeamHttpRequestParent
+  readonly #config: SeamHttpRequestConfig<TBody, TResponseKey>
+
+  constructor(
+    parent: SeamHttpRequestParent,
+    config: SeamHttpRequestConfig<TBody, TResponseKey>,
+  ) {
+    this.#parent = parent
+    this.#config = config
+  }
+
+  public get responseKey(): TResponseKey {
+    return this.#config.responseKey
+  }
+
+  public get url(): URL {
+    const { client } = this.#parent
+    const baseUrl = client.defaults.baseURL
+    if (baseUrl === undefined) {
+      throw new Error('baseUrl is required')
```

Comment:
Will this branch ever be taken in practice? (i.e. can we replace this with an `!` suffix on the preceding line?)

Reply:
If 	c is undefined or not a full URL, then we need to try and resolve it's full form using logic around `globalThis.location`. Then we would error out if we can't resolve a correct value, however in that case it means the class constructor validation failed to prevent this, so it should never happen.

If we don't want deal with this logic in this PR, instead of returning a URL, we can return `Pick<URL, 'path' | 'pathname' | 'searchParams' | 'query'>`, which only requires reading a path prefix from `baseUrl` if defined.